### PR TITLE
Update to Python 3.13.

### PR DESCRIPTION
**Pull request by fniessink**
Issues preventing update:

- [x] ~~No wheel available for lxml. Upstream issue: https://bugs.launchpad.net/lxml/+bug/2068828~~
- [x] ~~Building wheels for greenlet fails. Upstream PR: https://github.com/python-greenlet/greenlet/pull/396~~
- [x] ~~No wheel available for aiohttp. Upstream issue: https://github.com/aio-libs/aiohttp/issues/8194~~
- [x] ~~API-server unittest fails with `AttributeError: '_MainThread' object has no attribute '_tstate_lock'`. Gevent issue: https://github.com/gevent/gevent/issues/2054~~
- [x] ~~No `cimg/python` 3.13 image available: https://hub.docker.com/r/cimg/python/tags?page_size=&ordering=&name=3.13~~
- [x] ~~No `ghcr.io/astral-sh/uv:python3.13-bookworm` image available. Should be available shortly as the PR has been merged: https://github.com/astral-sh/uv/pull/8105~~
- [x] ~~ReadTheDocs does not support 3.13 yet: https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools-python, ticket: https://github.com/readthedocs/readthedocs.org/issues/11671~~

### CODE REVIEW

File: components/api_server/pyproject.toml [hunk 1]
```diff
@@ -36,7 +35,6 @@ optional-dependencies.tools = [
 ]
 
 [tool.ruff]
-target-version = "py312"
```
Comment (wkoot) [inline components/api_server/pyproject.toml:39]: Is there no added benefit to specifying a target version?

File: docs/src/conf.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 # -- Project information -----------------------------------------------------
 
 project = "Quality-time"
-copyright = "2021-2023, ICTU"  # noqa: A001
+copyright = "2021-2024, ICTU"  # noqa: A001
```
Comment (wkoot) [inline docs/src/conf.py:28]: Hehe, might as well set it to 2025 now or leave a ticket for next year ;)

File: components/api_server/pyproject.toml (hunk 1 above)
Comment (fniessink) [reply components/api_server/pyproject.toml:39]: It's (no longer?) needed: "If you're already using a pyproject.toml file, we recommend project.requires-python instead" (https://docs.astral.sh/ruff/settings/#target-version).

File: docs/src/conf.py (hunk 2 above)
Comment (fniessink) [reply docs/src/conf.py:28]: Done: https://github.com/ICTU/quality-time/issues/10354

**Pull request merged**